doc/cookbook/time_windows.md

```diff
@@ -136,54 +117,55 @@ Your second pipeline can then process these bins in parallel, via a glob pattern
+
+There is another pattern for moving time windows that avoids the binning of the above approach and maintains an up-to-date version of a moving time-windowed data set.  It also involves two pipelines:
 
 - *Pipeline 1* - Read in data, determine which files belong in your moving time window, and write the relevant files into an updated version of the moving time-windowed data set.  
 
 - *Pipeline 2* - Read in and analyze the moving time-windowed data set.
 
-**Note** - The disadvantage of this method is that "Pipeline 1" does not parallelize.  You would need to use a glob pattern of `/` in this pipeline to be able to extract the relevant files (corresponding to the moving time window) from all of the files that are input.
-
 Let's utilize our sales example again to see how this would work.  In the example, we want to keep a moving time window of the last three days worth of data.  Now say that our daily `sales` repo looks like the following:
 
 ```
 sales
-├── 01-01-17.csv
-├── 01-02-17.csv
-├── 01-03-17.csv
-└── 01-04-17.csv
+├── 01-01-17.json
+├── 01-02-17.json
+├── 01-03-17.json
+└── 01-04-17.json
 ```
 
-Our first pipeline would extract the last three days worth of data and put it into a repo corresponding to our moving time window (think of this as a "shuffle" step):
+When the January 4th file, `01-04-17.json`, is committed, our first pipeline pulls out the last three days of data and arranges it like so:
 
 ```
-moving_sales_window
-├── 01-02-17.csv
-├── 01-03-17.csv
-└── 01-04-17.csv
+last_three_days
+├── 01-02-17.json
+├── 01-03-17.json
+└── 01-04-17.json
 ```
 
-Then, when a new day's worth of data is added to our `sales` repo,
+Think of this as a "shuffle" step.  Then, when the January 5th file, `01-05-17.json`, is commited,  
 
 ```
 sales
-├── 01-01-17.csv
-├── 01-02-17.csv
-├── 01-03-17.csv
-├── 01-04-17.csv
-└── 01-05-17.csv
+├── 01-01-17.json
+├── 01-02-17.json
+├── 01-03-17.json
+├── 01-04-17.json
+└── 01-05-17.json
 ```
 
-the first pipeline would update the moving window:
+the first pipeline would again update the moving window:
 
 ```
-moving_sales_window
-├── 01-03-17.csv
-├── 01-04-17.csv
-└── 01-05-17.csv
+last_three_days
+├── 01-03-17.json
+├── 01-04-17.json
+└── 01-05-17.json
 ```
 
 Whatever analysis we need to run on the moving windowed data set in `moving_sales_window` can use a glob pattern of `/` or `/*` (depending on whether we need to process all of the time windowed files together or they can be processed in parallel).
 
+**Warning** - When creating this type of moving time-windowed data set, the concept of "now" or "today" is relative.  It is important that you make a sound choice for how to define time based on your use case (e.g., by defaulting to UTC).  If you have further questions about this issue, please do not hesitate to reach out to us via [Slack](http://slack.pachyderm.io/) or at support@pachyderm.io.
```

Comment:
It's worth making a note about why we split up the two steps. Step 1 can be completely parallelized with the datum as individual files. Every worker can just look at the file name and just either write it to the bin if it matches or do nothing if it doesn't. This can be blazingly fast if you also use the `lazy` field in the pipeline spec (link).

I just think it's worth mentioning because you could do it all in one step, but that wouldn't scale.